Tests for darling-core do not compile

`cargo build --tests` fails for `darling-core` with rustc 1.37.0:

```
$ cargo build --tests
   Compiling darling_core v0.10.0 (/home/robin/darling/core)
error[E0425]: cannot find value `idents` in this scope
  --> core/src/util/path_list.rs:85:13
   |
85 |             idents.to_strings(),
   |             ^^^^^^ not found in this scope

warning: trait objects without an explicit `dyn` are deprecated
   --> core/src/error/mod.rs:382:32
    |
382 |     fn cause(&self) -> Option<&StdError> {
    |                                ^^^^^^^^ help: use `dyn`: `dyn StdError`
    |
    = note: #[warn(bare_trait_objects)] on by default

error[E0599]: no method named `ok_or` found for type `std::result::Result<syn::Meta, syn::Error>` in the current scope
   --> core/src/from_meta.rs:390:32
    |
390 |         attribute.parse_meta().ok_or("Unable to parse".into())
    |                                ^^^^^

warning: use of deprecated item 'core::str::<impl str>::trim_left_matches': superseded by `trim_start_matches`
   --> core/src/options/shape.rs:137:20
    |
137 |         match word.trim_left_matches(self.prefix) {
    |                    ^^^^^^^^^^^^^^^^^ help: replace the use of the deprecated item: `trim_start_matches`
    |
    = note: #[warn(deprecated)] on by default

warning: use of deprecated item 'core::str::<impl str>::trim_right_matches': superseded by `trim_end_matches`
   --> core/src/options/shape.rs:191:34
    |
191 |             let ty = self.prefix.trim_right_matches('_');
    |                                  ^^^^^^^^^^^^^^^^^^ help: replace the use of the deprecated item: `trim_end_matches`

error[E0599]: no method named `ok_or` found for type `std::result::Result<syn::Meta, syn::Error>` in the current scope
  --> core/src/util/path_list.rs:73:32
   |
73 |         attribute.parse_meta().ok_or("Unable to parse".into())
   |                                ^^^^^

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0425, E0599.
For more information about an error, try `rustc --explain E0425`.
error: Could not compile `darling_core`.

To learn more, run the command again with --verbose.
```

--- a/core/src/from_meta.rs
+++ b/core/src/from_meta.rs
@@ -387,7 +387,7 @@ mod tests {
     /// parse a string as a syn::Meta instance.
     fn pm(tokens: TokenStream) -> ::std::result::Result<syn::Meta, String> {
         let attribute: syn::Attribute = parse_quote!(#[#tokens]);
-        attribute.parse_meta().ok_or("Unable to parse".into())
+        attribute.parse_meta().map_err(|_| "Unable to parse".into())
     }
 
     fn fm<T: FromMeta>(tokens: TokenStream) -> T {

--- a/core/src/util/path_list.rs
+++ b/core/src/util/path_list.rs
@@ -70,7 +70,7 @@ mod tests {
     /// parse a string as a syn::Meta instance.
     fn pm(tokens: TokenStream) -> ::std::result::Result<Meta, String> {
         let attribute: Attribute = parse_quote!(#[#tokens]);
-        attribute.parse_meta().ok_or("Unable to parse".into())
+        attribute.parse_meta().map_err(|_| "Unable to parse".into())
     }
 
     fn fm<T: FromMeta>(tokens: TokenStream) -> T {
@@ -82,7 +82,7 @@ mod tests {
     fn succeeds() {
         let paths = fm::<PathList>(quote!(ignore(Debug, Clone, Eq)));
         assert_eq!(
-            idents.to_strings(),
+            paths.to_strings(),
             vec![
                 String::from("Debug"),
                 String::from("Clone"),

